src/components/OptionList/components/Checkbox/Checkbox.scss

```diff
@@ -3,64 +3,145 @@
+        @include control-backdrop(focused);
+      }
+    }
+
+    &:disabled {
+      + .Backdrop {
+        @include control-backdrop(disabled);
+      }
+
+      ~ .Icon {
+        @include recolor-icon(color('ink', 'lightest'));
+      }
     }
   }
 
-  &:focus,
-  &:active {
-    + .Backdrop {
+  &.active {
+    .Backdrop {
       @include control-backdrop(focused);
     }
   }
 
-  &:disabled {
-    + .Backdrop {
-      @include control-backdrop(disabled);
+  .Backdrop {
+    @include control-backdrop;
+    position: absolute;
+    top: 0;
+    right: 0;
+    bottom: 0;
+    left: 0;
+  }
+
+  .Icon {
+    @include recolor-icon(color('indigo'));
+    position: absolute;
+    top: 50%;
+    left: 50%;
+    transform: translate(-50%, -50%) scale(0);
+    transform-origin: 50% 50%;
+    transition: control-icon-transition();
+  }
+}
+
+.Checkbox.newDesignLanguage {
+  .Input {
+    &.keyFocused {
+      + .Backdrop {
+        @include focus-ring($style: 'focused');
+      }
     }
 
-    ~ .Icon {
-      @include recolor-icon(color('ink', 'lightest'));
+    &:active:not(:disabled),
+    &:checked,
+    &.Input-indeterminate {
+      + .Backdrop {
+        @include control-backdrop(active, $global-theming: true);
+      }
+      ~ .Icon {
+        transition-timing-function: var(--p-ease-in);
+        transform: translate(-50%, -50%) scale(1);
+        opacity: 1;
+        transition: opacity var(--p-duration-1-5-0) var(--p-ease),
+          transform var(--p-duration-1-5-0) var(--p-ease);
+      }
+    }
+
+    &:disabled {
+      + .Backdrop {
+        @include control-backdrop(disabled, $global-theming: true);
+      }
+    }
+
+    &:disabled:checked {
+      + .Backdrop {
+        background: var(--p-border-secondary-disabled);
+        &::before {
+          background: var(--p-border-secondary-disabled);
+        }
+      }
+    }
+  }
+
+  &.active {
+    .Backdrop {
+      @include focus-ring($style: 'focused');
     }
   }
-}
 
-.active {
   .Backdrop {
-    @include control-backdrop(focused);
+    @include control-backdrop($global-theming: true);
+    position: relative;
+    display: block;
+    width: 100%;
+    height: 100%;
+    @include focus-ring($border-width: var(--p-control-border-width));
   }
-}
 
-.Backdrop {
-  @include control-backdrop;
-  position: absolute;
-  top: 0;
-  right: 0;
-  bottom: 0;
-  left: 0;
+  .Icon {
+    transform: translate(-50%, -50%) scale(0.25);
+    opacity: 0;
+    transition: opacity var(--p-duration-1-0-0) var(--p-ease),
+      transform var(--p-duration-1-0-0) var(--p-ease);
+  }
 }
 
 .Icon {
-  @include recolor-icon(color('indigo'));
+  @include recolor-icon(var(--p-icon-on-interactive, color('indigo')));
   position: absolute;
   top: 50%;
   left: 50%;
-  transform: translate(-50%, -50%) scale(0);
   transform-origin: 50% 50%;
-  transition: control-icon-transition();
+  pointer-events: none;
+
+  @media (-ms-high-contrast: active) {
+    fill: ms-high-contrast-color('text');
```

Comment:
Why not use `recolor-icon` here as well?

Reply:
not sure. Not something I touched. Wanted to test with virtual box which I just updated but it won't load :(